addons/calendar/models/calendar_attendee.py

```diff
@@ -0,0 +1,199 @@
+                trigger = valarm.add('TRIGGER')
+                trigger.params['related'] = ["START"]
+                if interval == 'days':
+                    delta = timedelta(days=duration)
+                elif interval == 'hours':
+                    delta = timedelta(hours=duration)
+                elif interval == 'minutes':
+                    delta = timedelta(minutes=duration)
+                trigger.value = delta
+                valarm.add('DESCRIPTION').value = alarm.name or 'Odoo'
+        for attendee in event_obj.attendee_ids:
+            attendee_add = event.add('attendee')
+            attendee_add.value = 'MAILTO:' + (attendee.email or '')
+        res = cal.serialize()
+        return res
+
+    @api.multi
+    def _send_mail_to_attendees(self, email_from=tools.config.get('email_from', False),
+                                template_xmlid='calendar_template_meeting_invitation', force=False):
+        """
+        Send mail for event invitation to event attendees.
+        @param email_from: email address for user sending the mail
+        @param force: If set to True, email will be sent to user himself. Usefull for example for alert, ...
+        """
+        res = False
+
+        if self.env['ir.config_parameter'].get_param('calendar.block_mail', default=False) or self.env.context.get("no_mail_to_attendees"):
+            return res
+
+        Mail = self.env['mail.mail']
+        local_context = self.env.context.copy()
+        color = {
+            'needsAction': 'grey',
+            'accepted': 'green',
+            'tentative': '#FFFF00',
+            'declined': 'red'
+        }
+
+        Template = self.env.ref('calendar.%s' % template_xmlid)
+        act_id = self.env.ref('calendar.view_calendar_event_calendar').id
+        local_context.update({
+            'color': color,
+            'action_id': self.env['ir.actions.act_window'].search([('view_id', '=', act_id)], limit=1).id,
+            'dbname': self._cr.dbname,
+            'base_url': self.env['ir.config_parameter'].get_param('web.base.url', default='http://localhost:8069')
+        })
+
+        for attendee in self:
+            if attendee.email and email_from and (attendee.email != email_from or force):
+                ics_file = self.get_ics_file(attendee.event_id)
+                mail_id = Template.with_context(local_context).send_mail(attendee.id)
+
+                vals = {}
+                if ics_file:
+                    vals['attachment_ids'] = [(0, 0, {'name': 'invitation.ics',
+                                                      'datas_fname': 'invitation.ics',
+                                                      'datas': str(ics_file).encode('base64')})]
+                vals['model'] = None  # We don't want to have the mail in the tchatter while in queue!
+                the_mailmess = Mail.browse(mail_id).mail_message_id
+                the_mailmess.write(vals)
+        mail_id = Mail.browse(mail_id)
```

Comment:
what if there is no mail_id, currently if I create new event and click on Send mail it throws traceback